Sources/Controllers/Map/Widgets/OATopCoordinatesWidget.m

```diff
@@ -0,0 +1,294 @@
+                    _latTextView.text = [OALocationConvert getLocationOlcName:lat lon:lon];
+                }
+                else
+                {
+                    _latTextView.hidden = NO;
+                    _lonTextView.hidden = NO;
+                    _latImageView.hidden = NO;
+                    _lonImageView.hidden = NO;
+                    _verticalSeparator.hidden = NO;
+                    
+                    NSString *coordinatesString = [OALocationConvert formatLocationCoordinates:lat lon:lon format:[self getConverterFormat:format]];
+                    NSArray<NSString *> *coordinates = [coordinatesString componentsSeparatedByString:@","];
+                    latText = coordinates[0];
+                    lonText = [coordinates[1] trim];
+                    
+                    NSString* latDayImg = lat >= 0 ? @"widget_coordinates_latitude_north_day" : @"widget_coordinates_latitude_south_day";
+                    NSString* latNightImg = lat >= 0 ? @"widget_coordinates_latitude_north_night" : @"widget_coordinates_latitude_south_night";
+                    NSString* lonDayImg = lon >= 0 ? @"widget_coordinates_longitude_east_day" : @"widget_coordinates_longitude_west_day";
+                    NSString* lonNightImg = lon >= 0 ? @"widget_coordinates_longitude_east_night" : @"widget_coordinates_longitude_west_night";
+                    
+                    //not a bug: in android in Night mode in this case shows Day icons too.
+                    [_latImageView setImage:[UIImage imageNamed:nightMode ? latDayImg : latNightImg]];
+                    [_lonImageView setImage:[UIImage imageNamed:nightMode ? lonDayImg: lonNightImg]];
+                    
+                    _latTextView.text = latText;
+                    _lonTextView.text = lonText;
+                }
+            }
+            else
+            {
+                _latTextView.hidden = NO;
+                _lonTextView.hidden = YES;
+                _latImageView.hidden = YES;
+                _lonImageView.hidden = YES;
+                _verticalSeparator.hidden = YES;
+                _latTextView.text = OALocalizedString(@"searching_gps");
+            }
+            
+            self.backgroundColor = UIColorFromRGB(nav_bar_night);
+            _latTextView.textColor = UIColorFromRGB(text_primary_night);
+            _lonTextView.textColor = UIColorFromRGB(text_primary_night);
+            _horisontalSeparator.hidden = !isPortrait;
+            
+            _lastUpdatingTime = [[NSDate new] timeIntervalSince1970];
+        }
+        
+        [self layoutSubviews];
+        return NO;
+    }
+    else
+    {
+        return NO;
+    }
+}
+
+- (int) getConverterFormat:(int)format
+{
+    return format + 101;
+}
+
+- (BOOL) shouldUpdate
+{
+    BOOL isFirstLaunch = _lastUpdatingTime == 0;
+    
+    BOOL isVisibilityChabged = !isFirstLaunch && _cachedVisibiliy != [_settings.showCoordinatesWidget get];
```

Comment:
isVisibilityChabged > isVisibilityChanged
isFirstLaunch is not related to visibility change.